dotnet/src/SemanticKernel.Skills/Skills.Document/DocumentSkill.cs

```diff
@@ -112,4 +112,19 @@ public async Task AppendTextAsync(string text, SKContext context)
 
         await stream.DisposeAsync();
     }
+
+    /// <summary>
+    /// Recursively list all documents under a directory.
+    /// </summary>
+    /// <param name="directoryPath">Path of directory under which to list documents.</param>
+    /// <param name="context">Semantic kernel context</param>
+    /// <returns>An array of the paths of the documents under the given directory</returns>
+    [SKFunction("Recursively list all documents under a directory.")]
+    [SKFunctionInput(Description = "Path of directory under which to list documents.")]
+    public Task<string[]> RecursivelyListDocumentsUnderDirectoryAsync(string directoryPath, SKContext context)
+    {
+        this._logger.LogInformation("Listing documents under {0}", directoryPath);
+
+        return this._fileSystemConnector.RecursivelyListFilesUnderDirectoryAsync(directoryPath, context.CancellationToken);
```

Comment:
await keyword is missing.

Reply:
Not needed here.  I'm just passing the Task up to the caller and the caller will do the await.